### PR TITLE
Private/poonam/vmha sync

### PR DESCRIPTION
**Pull request by poonam-agarwal28**

### CODE REVIEW

File: hamgr/hamgr/db/api.py [hunk 1]
```diff
@@ -155,7 +182,20 @@ def dbsession():
...
+    try:
+        yield mdb_session
+        if _has_unsaved_changes(mdb_session):
+            mdb_session.commit()
+    except SQLAlchemyError as se:
+        LOG.error('Error working with db session: %s', se)
+        if _has_unsaved_changes(mdb_session):
+            mdb_session.rollback()
+    finally:
+        mdb_session.close()
```
Comment (poojaghumre) [inline hamgr/hamgr/db/api.py:198]: nit: pleas add new line after this func

File: hamgr/hamgr/db/api.py [hunk 2]
```diff
@@ -331,6 +371,19 @@ def create_change_event(cluster_id, events, event_id=''):
...
 
 
+def get_hosts_by_name(name):
+    if name is None:
+        raise exceptions.ArgumentException('host name is empty')
+    with mdbsession() as session:
+        try:
+            query = session.query(Hosts)
+            query = query.filter_by(name=name, deleted=0)
+            result = query.first()
```
Comment (poojaghumre) [inline hamgr/hamgr/db/api.py:381]: lets log a warning if there are more than 1 hosts here?

File: hamgr/hamgr/providers/nova.py [hunk 3]
```diff
@@ -521,6 +523,41 @@ def process_host_aggregate_changes(self):
                 # if more hosts in nova aggregate, means they exist in nova aggregate but not in masakari
                 # if more hosts in masakari, means some of they are removed from nova aggregate
 
+
```
Comment (poojaghumre) [inline hamgr/hamgr/providers/nova.py:526]: nit: remove extra line

**Review by poojaghumre — CHANGES_REQUESTED** [review]

File: hamgr/hamgr/db/api.py (hunk 2 above)
Comment (poonam-agarwal28) [reply hamgr/hamgr/db/api.py:381]: That shouldnt be a possibility I presume ? This function is only checking if the host is already apart of Masakari Database and will add only when it doesnt find an existing entry

File: hamgr/hamgr/providers/nova.py (hunk 3 above)
Comment (poonam-agarwal28) [reply hamgr/hamgr/providers/nova.py:526]: OK will do

**Pull request merged**